src/uthenticode.cpp

```diff
@@ -541,36 +551,53 @@ std::string calculate_checksum(peparse::parsed_pe *pe, checksum_kind kind) {
   pe_bits.erase(pe_bits.begin() + cert_table_offset, pe_bits.begin() + cert_table_offset + 8);
   pe_bits.erase(pe_bits.begin() + pe_checksum_offset, pe_bits.begin() + pe_checksum_offset + 4);
 
-  /* Hash pe_bits, which contains the rest of the header.
+  /* Build up the list of sections in the PE, in ascending order by PointerToRawData
+   * (i.e., by file offset).
+   *
+   * NOTE(ww): Ideally we'd use a capture with the C++ lambda here, but C++ lambdas can't be
+   * used within C callbacks unless they're captureless.
    */
-  const auto *md = EVP_get_digestbynid(nid);
-  auto *md_ctx = EVP_MD_CTX_new();
-  EVP_DigestInit(md_ctx, md);
-  EVP_DigestUpdate(md_ctx, pe_bits.data(), pe_bits.size());
-
-  if (security_dir.VirtualAddress > 0) {
-    /* If a certificate table exists, hash everything before and after it.
-     */
-    EVP_DigestUpdate(md_ctx,
-                     pe->fileBuffer->buf + size_of_headers,
-                     security_dir.VirtualAddress - size_of_headers);
```

Comment:
Have you removed this part?

Reply:
Sort of -- part of this is covered by the section iteration, but not all of it (which is why we've regressed on some PEs, as noted in https://github.com/trailofbits/uthenticode/pull/84#issuecomment-1660794391).

I need to think more about an appropriate way to do this -- we need to perform the sorted section iteration *and* hash everything after the sections.